test/ginkgo-ext/scopes.go

```diff
@@ -116,6 +119,33 @@ func init() {
 	commandFlags.Parse(args)
 }
 
+func (s *scope) isUnset() bool {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
+	return (s.counter == -1)
+}
+
+func (s *scope) isZero() bool {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
+	return (s.counter == 0)
+}
+
+func (s *scope) setSafely(val int) {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
+	s.counter = int32(val)
+}
+
+func (s *scope) decrementSafely() {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
+	s.counter--
+	if s.counter < 0 {
+		panic("unexpected negative scope counter value")
```

Comment:
@nebril so from the Jenkins logs it looks we do actually hit this, and my understanding was that code was assuming it doesn't actually go below zero - WDYT?

Reply:
I added the actual value here also, so we can see if it's `-1` or `-99` :)